File: Python/Leetcode 2306.py
class Solution:
    def distinctNames(self, ideas: List[str]) -> int:
        Vect = [set() for _ in range(26)]
        # print(Vect)

        for string in ideas:
            Vect[ord(string[0]) - ord('a')].add(string[1:])

        # print(Vect)
        answer = 0
        for i in range(0,25):
            

            j = i + 1
            while j < 26:
                set1 = set()
                set1.update(Vect[i])
                set1.update(Vect[j])
                print(set1)
                answer = answer + (len(set1) - len(Vect[i]))*(len(set1) - len(Vect[j]))
                j = j + 1
            # print(answer)
        return answer*2

File: C++/Leetcode 2306.cpp
#include <stdio.h>
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    long long distinctNames(vector<string>& ideas) {
        vector<unordered_set<string>> vect(26);

        for (int i = 0; i < ideas.size(); i++)
        {
            // All the word starting with the same letter are been compiled under the same letter
            vect[ideas[i][0] - 'a'].insert(ideas[i].substr(1));
        }

        long long answer = 0;
        for (int i = 0; i < 25; i++)
        {
            for (int j = i + 1; j <= 25; j++)
            {
                unordered_set<string> set1;

                set1.insert(vect[i].begin(),vect[i].end());
                set1.insert(vect[j].begin(),vect[j].end());

                answer += (set1.size() - vect[i].size())*(set1.size() - vect[j].size());
            }
        }
        return answer*2;

    }
};